pkg/api/login.go

```diff
@@ -57,18 +57,27 @@ func (hs *HTTPServer) LoginView(c *models.ReqContext) {
 		return
 	}
 
-	if !c.IsSignedIn {
-		c.HTML(200, ViewIndex, viewData)
-		return
-	}
+	// If we already are signed in (Via auth proxy for example)
+	if c.IsSignedIn {
+		// Create login token
+		hs.loginUserWithUser(&models.User{
+			Id:    c.SignedInUser.UserId,
+			Email: c.SignedInUser.Email,
+			Login: c.SignedInUser.Login,
+		}, c)
+
+		if redirectTo, _ := url.QueryUnescape(c.GetCookie("redirect_to")); len(redirectTo) > 0 {
```

Comment:
Think this code should still execute last in this method always when logged in and by that having old `if !c.IsSignedIn {` code block before auth proxy enabled and token login. 

Like this:
- Not logged in? Return login page
  - What to do here if auth proxy is enabled? We don't want to render the login form/page right.
- Auth proxy enabled? Auth token login
- Redirect logic